pineappl/src/grid.rs

```diff
@@ -263,6 +264,9 @@ pub enum GridError {
         /// Maximum supported file format version for this library.
         supported_version: u64,
     },
+    /// Returned from [`Grid::evolve`] if the evolution failed.
+    #[error("failed to evolve grid: {0}")]
+    EvolutionFailure(String),
```

Comment:
If moving all the evolution bits is accepted, in principle also this one should be moved for consistency.

Now, it is a bit uncomfortable, because part of an `enum`, and
> practicality beats purity
but there would also be a proper solution for this, as explained in [this answer](https://stackoverflow.com/a/25214677/8653979).

Reply:
I will not touch this for the time being as I'm planning to change it completely in #85.